content/docs/cookbooks/tls-auth.md

```diff
@@ -0,0 +1,46 @@
+---
+title: Securing Prometheus endpoints using TLS authentication
+sort_rank: 1
+---
+
+# Securing Prometheus endpoints using TLS auth
+
+Prometheus does *not* directly support authentication via [Transport Layer Security](https://en.wikipedia.org/wiki/Transport_Layer_Security), and this is so [by design](/docs/introduction/faq/#tls). In fact, Prometheus provides no server-side authentication, authorization, or encryption.
+
+Nonetheless, we do strongly recommended that you use TLS or some other form of authentication when using Prometheus by applying it via reverse proxy. You can use any proxy you like with Prometheus. In this cookbook, however, we'll cover only [nginx](#nginx).
+
+NOTE: For more information on the reasoning behind these and other security-related decisions in Prometheus, see the [Security Model](/docs/operating/security) documentation.
+
+## nginx example
+
+CAUTION: The nginx configuration below is just an example. We strongly recommend consulting the official [nginx documentation](http://nginx.org/en/docs/http/ngx_http_ssl_module.html#ssl_certificate) when creating a customized configuration, deciding on TLS protocols and ciphers to use, and so on.
+
+Let's say that you're running Prometheus behind an [nginx](https://www.nginx.com/) server running at https://www.my-nginx-proxy.org and you want to access Prometheus via the `/prometheus` endpoint. Here's an example cURL command that would query the Prometheus server's `/metrics` endpoint:
+
+```bash
+$ curl https://my-nginx-proxy.org/prometheus/metrics
+```
+
+Let's also say that you've generated the following using [OpenSSL](https://www.openssl.org/) or an analogous tool:
+
+* an SSL certificate at `/root/certs/my-nginx-proxy.com/my-nginx-proxy.com.crt`
+* an SSL key at `/root/certs/my-nginx-proxy.com/my-nginx-proxy.com.key`
+
+Below is an example nginx configuration that would handle TLS authentication at the nginx level and then proxy all connections on port 443 to a Prometheus server running on the same host:
```

Comment:
This configuration does not proxy all connections to Prometheus, only those requesting /prometheus